lib/rules/number.js

```diff
@@ -0,0 +1,175 @@
+import SyncRule from '../types/syncRule';
+
+class NumberValidator extends SyncRule {
+
+    validateSync(params) {
+        return typeof params.value === 'number' &&
+        !Number.isNaN(params.value) &&
+        Number.isFinite(params.value);
+    }
+
+    static ruleName() {
+        return 'number';
+    }
+}
+
+class Integer extends SyncRule {
+
+    validateSync(params) {
+        return Number.isInteger(params.value);
+    }
+
+    static ruleName() {
+        return 'number.integer';
+    }
+}
+
+class Min extends SyncRule {
+
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value >= this._val || {
+            min: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'number.min';
+    }
+}
+
+class Max extends SyncRule {
+
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value <= this._val || {
+            max: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'number.max';
+    }
+}
+
+class Greater extends SyncRule {
+
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value > this._val || {
+            greater: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'number.greater';
+    }
+}
+
+class Less extends SyncRule {
+
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value < this._val || {
+            less: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'number.less';
+    }
+}
+
+class Negative extends SyncRule {
+
+    validateSync(params) {
+        return params.value < 0;
+    }
+
+    static ruleName() {
+        return 'number.negative';
+    }
+}
+
+class Positive extends SyncRule {
+
+    validateSync(params) {
+        return params.value > 0;
+    }
+
+    static ruleName() {
+        return 'number.positive';
+    }
+}
+
+class Multiple extends SyncRule {
+
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value % this._val === 0 || {
+            multiple: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'number.multiple';
+    }
+}
+
+const precisionRegEx = /^-?\d+(\.\d+)?$/;
+
+class Precision extends SyncRule {
+
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    coerce(value) {
+        return value.toFixed(this._val);
```

Comment:
Jojen converts by default, if you add the `convert: false` option it will fail as you expect.